internal/service/transactions.go

```diff
@@ -19,91 +19,177 @@ package service
 import (
 	"context"
 	"fmt"
+	"math/big"
+	"strings"
+	"time"
 
-	ledgererr "github.com/coinbase-samples/ib-ledger-go/internal/errors"
+	"github.com/amzn/ion-go/ion"
+	"github.com/coinbase-samples/ib-ledger-go/internal/config"
+	"github.com/coinbase-samples/ib-ledger-go/internal/model"
+	"github.com/coinbase-samples/ib-ledger-go/internal/qldb"
 	"github.com/coinbase-samples/ib-ledger-go/internal/utils"
 	api "github.com/coinbase-samples/ib-ledger-go/pkg/pbs/ledger/v1"
+	"github.com/google/uuid"
 	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
 
 	"google.golang.org/grpc/codes"
-	"google.golang.org/protobuf/types/known/timestamppb"
+	"google.golang.org/grpc/status"
 )
 
 func (s *Service) CreateTransaction(ctx context.Context, req *api.CreateTransactionRequest) (*api.CreateTransactionResponse, error) {
 	l := ctxlogrus.Extract(ctx)
 
 	if err := req.ValidateAll(); err != nil {
 		l.Debugf("invalid create transaction request: %v", req)
-		return nil, fmt.Errorf("ib-ledger-go: %w", err)
+		return nil, handleValidationError(err)
+	}
+
+	orderAmount := new(big.Int)
```

Comment:
This format is easier to read and works the same

	var orderAmount big.Int
	orderAmount.SetString("1000", 10)